### PR TITLE
☠☕ Decaffeinate

### PR DESCRIPTION
**Pull request by jasonrudolph**
### TODO

- [x] Decaffeinate `lib/reporter.coffee`
- [x] Decaffeinate `lib/metrics.coffee`
- [x] Replace CoffeeScript linting config with [standard](https://standardjs.com/) linting config

### Verification Process

See https://github.com/atom/metrics/pull/88#issuecomment-396436950 and https://github.com/atom/metrics/pull/88#issuecomment-396609209.

/fyi @annthurium

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -21,6 +21,19 @@
     "node-uuid": "~1.4.7"
   },
   "devDependencies": {
-    "coffeelint": "^1.9.7"
+    "standard": "*"
+  },
+  "standard": {
```
Comment (50Wliu) [inline package.json]: To match other Atom packages that use standard, this should use the following block instead:
```
"standard": {
    "env": [
      "browser",
      "node",
      "atomtest",
      "jasmine"
    ],
    "globals": [
      "atom"
    ]
  },
```
See for example https://github.com/atom/open-on-github/blob/58899b80b73f26dbe75e9938a61dfd3b8690de9e/package.json#L32-L42

**Comment by thomasjo** [comment]
> re smoke testing: is there a way to record network calls made from Atom? If so, you could compare the calls to Google Analytics when performing the same set of actions, before and after these changes

@annthurium That's a really great idea! It's very easy to do in a (semi-)manual manner using e.g [Wireshark](https://www.wireshark.org/).

**Comment by annthurium** [comment]
okay, so I did some manual smoke testing.  I did this by looking at the network tab within the Chrome console in Atom, because Wireshark is kind of a pain when every app is constantly phoning home. :-)

tested the following actions on both the coffeescript and decaffeinated versions:
- opening a file (with command t) sends a "file open" event
- opening a file (by opening a new pane) sends a "file open" event
- opting out of metrics sends an opt-out event
- opting in to metrics sends an opt-in event
- refreshing Atom windows sends a "window ended" event.  
- refreshing Atom window sends a "window started" event.
- resfreshing Atom window sends timing/latency data.
- deprecation.  `grim` reports on packages that use old (< 1.0) atom apis.  It took awhile to find a package that actually triggers a `grim` deprecation warning.  I eventually looked in Google Analytics at real time deprecation warnings and was able to find one (atom-terminal-panel@4.4.4) and verify that the deprecation event is sent correctly.

possible issues, or things I wasn't confident I tested properly
- I don't understand the `watchCommands` code. It looks like we're filtering out all commands that have a colon.  Allll the commands in the command palette have colons.  So what is this actually doing?

**Comment by thomasjo** [comment]
> I don't understand the watchCommands code. It looks like we're filtering out all commands that have a colon. Allll the commands in the command palette have colons. So what is this actually doing?

@annthurium I don't blame you; the [original code](https://github.com/atom/metrics/blob/f9bf99272d07fae3c40cb87e2cc60ff49e67d967/lib/metrics.coffee#L93) was somewhat obtuse; it returns unless the result of `indexOf(':')` is greater than `-1`, which means it returns unless the event name contains a colon. In other words it filters out all events that are _not_ commands. The expression would have been much easier to read if we had used a normal `if` as opposed to an `unless`. The [new code](https://github.com/atom/metrics/blob/d0fb09af2c6774342613f4da4d95afd1808df306/lib/metrics.js#L123) is even less readable, and really should be rewritten to something like
```js
if (eventName.indexOf(':') === -1) return
```

Hope my explanation was helpful.

File: package.json (hunk 1 above)
Comment (jasonrudolph) [reply package.json]: Thanks, @50Wliu. Updated in 023e95a.

File: lib/metrics.js [hunk 2]
```diff
@@ -120,7 +120,7 @@ module.exports = {
 
       if (eventName.startsWith('core:') || eventName.startsWith('editor:')) return
 
-      if (!(eventName.indexOf(':') > -1)) return
+      if (!eventName.includes(':')) return
```
Comment (thomasjo) [inline lib/metrics.js]: :heart_eyes:

Comment (jasonrudolph) [reply lib/metrics.js]: @thomasjo: Thanks for [pointing out](https://github.com/atom/metrics/pull/88#issuecomment-396555060) this opportunity for improvement! :zap:

**Comment by annthurium** [comment]
thank you @thomasjo @jasonrudolph !!

**Comment by damieng** [comment]
Sorry I didn't send the notes about testing sooner. There was a or that
switched third simply over to fetch which displayed all the requests in the
network tab

**Pull request merged**